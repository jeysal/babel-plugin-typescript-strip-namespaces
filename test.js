const { transform } = require('@babel/core');
const plugin = require('.');

test('strips away a namespace containing types and interfaces', () => {
  const { code } = transform(
    `
    namespace N {
      type A = string;
      interface B {
        x: string;
      }
    }

    class N {}
    `,
    { plugins: [plugin, '@babel/plugin-syntax-typescript'] }
  );

  expect(code).toEqual(`class N {}`);
});

test('allows export declarations', () => {
  const { code } = transform(
    `
    namespace N {
      export type A = string;
    }

    class N {}
    `,
    { plugins: [plugin, '@babel/plugin-syntax-typescript'] }
  );

  expect(code).toEqual(`class N {}`);
});

test('handles nested namespaces', () => {
  const { code } = transform(
    `
    namespace N {
      type A = string;
      namespace Inner {
        type B = string;
      }
    }

    class N {}
    `,
    { plugins: [plugin, '@babel/plugin-syntax-typescript'] }
  );

  expect(code).toEqual(`class N {}`);
});

test('fails if a namespace contains anything else', () => {
  expect(() =>
    transform(
      `
      namespace N {
         const x = 42;
      }
      `,
      {
        plugins: [plugin, '@babel/plugin-syntax-typescript'],
        highlightCode: false,
      }
    )
  ).toThrowErrorMatchingSnapshot();
});

test('fails if a nested namespace contains anything else', () => {
  expect(() =>
    transform(
      `
      namespace N {
        type A = string;
        namespace Inner {
          const x = 42;
        }
      }
      `,
      {
        plugins: [plugin, '@babel/plugin-syntax-typescript'],
        highlightCode: false,
      }
    )
  ).toThrowErrorMatchingSnapshot();
});

test('works with plugin-transform-typescript', () => {
  const { code } = transform(
    `
    namespace N {
      type A = string;
      interface B {
        x: string;
      }
    }

    class N {}
    `,
    {
      plugins: ['@babel/plugin-transform-typescript', plugin], // even in the wrong order!
    }
  );
  expect(code).toEqual('class N {}');
});
