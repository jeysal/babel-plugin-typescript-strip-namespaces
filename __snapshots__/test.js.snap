// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fails if a namespace contains anything else 1`] = `
"undefined: Namespaces must only contain type and interface declarations
  1 | 
  2 |       namespace N {
> 3 |          const x = 42;
    |          ^
  4 |       }
  5 |       "
`;

exports[`fails if a nested namespace contains anything else 1`] = `
"undefined: Namespaces must only contain type and interface declarations
  3 |         type A = string;
  4 |         namespace Inner {
> 5 |           const x = 42;
    |           ^
  6 |         }
  7 |       }
  8 |       "
`;
